api/v1alpha1/openapi.yaml

```diff
@@ -3009,3 +3009,157 @@ components:
       - 'TemplateDiscriminatorKubernetesSec'
       - 'TemplateDiscriminatorInlineConfig'
       - 'TemplateDiscriminatorHttpConfig'
+    CertSigningRequestApproval:
+      type: object
+      properties:
+        labels:
+          type: object
+          additionalProperties:
+            type: string
+          description: 'labels is a set of labels to apply to the device.'
+        approved:
+          type: boolean
+          description: 'approved indicates whether the request has been approved.'
+        approvedBy:
+          type: string
+          description: 'approvedBy is the name of the approver.'
+        approvedAt:
+          type: string
+          format: date-time
+          description: 'approvedAt is the time at which the request was approved.'
+      required:
+        - approved
+    CertSigningRequestCondition:
+    # this could be replaced with the pre-existing Condition schema, but Condition is not a 1:1 match to the k8s schema
+      description: 'Information about a CSR'
+      properties:
+        lastTransitionTime:
+          description: 'Time of the last status transition'
+          type: string
+          format: date-time
+        lastUpdateTime:
+          description: 'Time of the last update to current condition'
+          type: string
+          format: date-time
+        message:
+          description: 'Human readable message with details about the request state'
+          type: string
+        reason:
+          description: 'Brief reason for the request state'
+          type: string #TODO - enum?
+        status:
+          description: 'True, False, or Unknown'
+          $ref: '#/components/schemas/ConditionStatus'
+        type:
+          description: 'Approved, Denied, and/or Failed'
+          type: string
+          enum:
+            - 'Approved'
+            - 'Denied'
+            - 'Failed'
```

Comment:
question: what state is the request in if it has not been yet been approved?

Reply:
Good thought - it's modeled on https://raw.githubusercontent.com/kubernetes/kubernetes/master/api/openapi-spec/swagger.json and they only have these 3 states, but we may want to include another that is something like "pending." This is part of the description from k8s: 

```
certificate is populated with an issued certificate by the signer after an Approved condition is present. This field is set via the /status subresource. Once populated, this field is immutable.\n\nIf the certificate signing request is denied, a condition of type \"Denied\" is added and this field remains empty. If the signer cannot issue the certificate, a condition of type \"Failed\" is added and this field remains empty"
```